virtual_ecosystem/models/animals/animal_cohorts.py

```diff
@@ -711,3 +705,33 @@ def inflict_natural_mortality(
         )
 
         self.die_individual(number_of_deaths, carcass_pool)
+
+    def migrate_juvenile_probability(self) -> float:
```

Comment:
The output of this function is not really a probability, but rather a flag if it should migrate or not based on the calculate probability. Both the docstring explaining the function, the return type (I'm surprised `mypy` didn't picked this) and the function name itself should be updated to reflect this. Maybe calling it `juvenile_should_migrate` or something like that to reflect that is actually a boolean check.

Or return the actual probability and then use `random.random()` in the calling function, which I think makes more sense.

Reply:
Oh good point! I'm surprised that didn't get picked up.